Assets/Scripts/MusicManager.cs

```diff
@@ -0,0 +1,222 @@
+    /// </summary>
+    private int QuestsCompleted
+    {
+        get { return questsCompleted_UseProperty; }
+        set
+        {
+            questsCompleted_UseProperty = value;
+            switch(currentMusicState)
+            {
+                case TownMusicState.Cursed:
+                    if (QuestsCompleted >= 2)
+                        isMusicFading = true;
+                    break;
+                case TownMusicState.Dreary:
+                    if(QuestsCompleted >= 5)
+                        isMusicFading = true;
+                    break;
+                case TownMusicState.Light:
+                    if(QuestsCompleted >= 12)
+                        isMusicFading = true;
+                    break;
+            }
+            if(isMusicFading)
+            {
+                isFadingOut = true;
+                currentMusicState++;
+                fadeStartTime = Time.time;
+            }
+        }
+    }
+
+    private CurrentMusic CurrentlyPlayingMusic
+    {
+        get { return currentlyPlayingMusic_UseProperty; }
+        set
+        {
+            if(value != currentlyPlayingMusic_UseProperty)
+            {
+                isMusicFading = true;
+                isFadingOut = true;
+                fadeStartTime = Time.time;
+                currentlyPlayingMusic_UseProperty = value;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Standard int for use with property
+    /// </summary>
+    private int questsCompleted_UseProperty;
+
+    /// <summary>
+    /// CurrentMusic variable for use with property
+    /// </summary>
+    private CurrentMusic currentlyPlayingMusic_UseProperty;
+
+    /// <summary>
+    /// The AudioSource component attached to the object
+    /// </summary>
+    private AudioSource audioSource;
+
+    /// <summary>
+    /// Is the music fading?
+    /// </summary>
+    private bool isMusicFading;
+
+    /// <summary>
+    /// Is the music fading out?
+    /// </summary>
+    private bool isFadingOut;
+
+    /// <summary>
+    /// When did the fade start?
+    /// </summary>
+    private float fadeStartTime;
+
+    /// <summary>
+    /// The volume of the AudioSource, as set in the editor
+    /// </summary>
+    private float startingVolume;
+
+    /// <summary>
+    /// What is the current music state, based on quests completed?
+    /// </summary>
+    private TownMusicState currentMusicState = TownMusicState.Cursed;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        audioSource = GetComponent<AudioSource>();
+        startingVolume = audioSource.volume;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        CheckQuestsCompleted();
+        if (isMusicFading)
+            FadeMusic();
+    }
+
+    /// <summary>
+    /// Check the quests completeed variable from dialogue system and update it locally
+    /// </summary>
+    private void CheckQuestsCompleted()
```

Comment:
I think you can create listeners / subscribe to events and check less frequently than update, isn't there a way to do that?

Reply:
I checked and didn't find a good way to check at relevant times. The closest I think I could get would be calling the function when a conversation ends, and I like being able to change the music in the middle of a conversation.